pkg/assembler/graphdb.go

```diff
@@ -0,0 +1,148 @@
+//
+// Copyright 2022 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package assembler
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/neo4j/neo4j-go-driver/v4/neo4j"
+	"github.com/guacsec/guac/pkg/assembler/graphdb"
+)
+
+// Note: This module is experimental and might change often!
+
+// StoreSubgraph stores a Graph to the graph database given by Client
+func StoreGraph(g Graph, client graphdb.Client) error {
```

Comment:
hmm, thinking if this would be a good candidate for a function receiver for `graphdb.Client`, so defining an interface for `GraphDB` and doing it as function receivers in  `github.com/guacsec/guac/pkg/assembler/graphdb/neo4j`?

Reply:
This is a possibility but has a potential issue that it would cause a dependency cycle between `graphdb/*.go` and `nodes.go` (or similar). I'd say, let's try this in a separate PR?